mono/utils/mono-threads-posix-signals.c

```diff
@@ -193,6 +194,9 @@ suspend_signal_handler (int _dummy, siginfo_t *info, void *context)
 	*/
 	pthread_sigmask (SIG_BLOCK, &suspend_ack_signal_mask, NULL);
 
+	/* ensure neither the compiler, neither the CPU reorders any memory operation */
+	mono_memory_barrier ();
```

Comment:
Is this really needed ? Semaphore posts (which is done in notify below) should include a memory barrier.